### DISCUSSION TITLE
Using the tag "destination=*" in routing

### DISCUSSION THREAD

**Discussion by ReinerMeyer** [osmandapp/OsmAnd#13493] [General]
Hi,

I use OSMAND for my autoroutes and hope that my request is correctly placed here.

I have recently tried to adapt the OSM-tag destination=* in its various forms to the road situation, with the hope that OSMAND also takes this into account in the acoustic announcement. However, this does not seem to be the case for every form of the destination tag.
My question: Which of these destination tags is taken into account by OSMAND?
destination
destination:to
destination:forward
destination:backward
destination:lanes
destination:lanes:forward
destination:lanes:backward

**Comment by tommycrock** [comment]
I think just destination and with :forward and :backward as well as :ref but would be interested to hear.
Some links:
https://wiki.openstreetmap.org/wiki/Key:destination
https://wiki.openstreetmap.org/wiki/User:Mueschel/DestinationTagging

**Comment by ReinerMeyer** [comment]
Thanks for the answer. But I'm interested in what OSMAND realy uses. May-be there are routing-scripts what I can check. Where I would find theses?

**Reply by tommycrock** [reply]
I would also like an answer from the developers, but something like this may help https://github.com/search?q=org%3Aosmandapp+destination%3Aforward&type=code

**Comment by ReinerMeyer** [comment]
Hi tommycrock, thanks again for this helpful answer.
More and more the tag destination:lanes is in use and it seems that this tag is not applied in OSMANS routing.
I am interested in the reason why this tag is not yet evaluated. Who gain give me an answer?

Another issue concerns to highway=stop in combination with direction=backward (e.g. node 5773774412). This traffic sign is in OSMAND applied in both directions. Where can I deposit this issue?

**Comment by vshcherb** [comment]
I'm pretty sure it's used
```
destination
destination:forward
destination:backward
```
Not used `destination:lanes`, we don't have complicated widgets display yet. For simple widget text it's not needed cause we take ref/name/destination for the road after the junction

**Reply by Lee-Carre** [reply]
Good to know; I've been considering tagging local roads with these keys.

**Reply by tommycrock** [reply]
Also `destination:ref` is used and not deduplicated from `ref`. `destination:to` and `destination:ref:to` are not used (these from my experience).
